### PR TITLE
KAFKA-16331: Remove EOSv1 from Kafka Streams integration tests

### PR DESCRIPTION
**Pull request by mjsax**
We will remove EOSv1 in 4.0, and thus can remove the corresponding integration tests.

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/EOSUncleanShutdownIntegrationTest.java [hunk 1]
```diff
@@ -71,8 +69,8 @@ public static void startCluster() throws IOException {
         CLUSTER.start();
         STREAMS_CONFIG.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
         STREAMS_CONFIG.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
-        STREAMS_CONFIG.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass());
-        STREAMS_CONFIG.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        STREAMS_CONFIG.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
+        STREAMS_CONFIG.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/EOSUncleanShutdownIntegrationTest.java:73]: Side improvement

File: streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java [hunk 2]
```diff
@@ -176,13 +176,13 @@ public void createTopics() throws Exception {
     }
 
     @ParameterizedTest
-    @ValueSource(strings = {StreamsConfig.AT_LEAST_ONCE, StreamsConfig.EXACTLY_ONCE, StreamsConfig.EXACTLY_ONCE_V2})
+    @ValueSource(strings = {StreamsConfig.AT_LEAST_ONCE, StreamsConfig.EXACTLY_ONCE_V2})
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java:179]: Why do we run this "eos test" with at-least-once ?

File: streams/src/test/java/org/apache/kafka/streams/integration/GlobalKTableEOSIntegrationTest.java [hunk 3]
```diff
@@ -186,7 +183,9 @@ public void shouldKStreamGlobalKTableLeftJoin(final String eosConfig) throws Exc
         TestUtils.waitForCondition(
             () -> {
                 globalState.clear();
-                replicatedStore.all().forEachRemaining(pair -> globalState.put(pair.key, pair.value));
+                try (final KeyValueIterator<Long, String> it = replicatedStore.all()) {
+                    it.forEachRemaining(pair -> globalState.put(pair.key, pair.value));
+                }
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/GlobalKTableEOSIntegrationTest.java:188]: Side improvement (more of the same below)

File: streams/src/test/java/org/apache/kafka/streams/integration/JoinGracePeriodDurabilityIntegrationTest.java [hunk 4]
```diff
@@ -91,9 +91,8 @@ public static void closeCluster() {
     private static final Serde<String> STRING_SERDE = Serdes.String();
     private static final long COMMIT_INTERVAL = 100L;
 
-    @SuppressWarnings("deprecation")
     @ParameterizedTest
-    @ValueSource(strings = {StreamsConfig.AT_LEAST_ONCE, StreamsConfig.EXACTLY_ONCE, StreamsConfig.EXACTLY_ONCE_V2})
+    @ValueSource(strings = {StreamsConfig.AT_LEAST_ONCE, StreamsConfig.EXACTLY_ONCE_V2})
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/JoinGracePeriodDurabilityIntegrationTest.java:95]: Why are we running this test parameterized?

File: streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java [hunk 5]
```diff
@@ -67,8 +67,8 @@ public class KTableSourceTopicRestartIntegrationTest {
     public static void startCluster() throws IOException {
         CLUSTER.start();
         STREAMS_CONFIG.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
-        STREAMS_CONFIG.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
-        STREAMS_CONFIG.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+        STREAMS_CONFIG.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
+        STREAMS_CONFIG.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java:71]: Side improvement

File: streams/src/test/java/org/apache/kafka/streams/integration/ResetPartitionTimeIntegrationTest.java [hunk 6]
```diff
@@ -88,9 +88,8 @@ public static void closeCluster() {
     private static final int DEFAULT_TIMEOUT = 100;
     private static long lastRecordedTimestamp = -2L;
 
-    @SuppressWarnings("deprecation")
     @ParameterizedTest
-    @ValueSource(strings = {StreamsConfig.AT_LEAST_ONCE, StreamsConfig.EXACTLY_ONCE, StreamsConfig.EXACTLY_ONCE_V2})
+    @ValueSource(strings = {StreamsConfig.AT_LEAST_ONCE, StreamsConfig.EXACTLY_ONCE_V2})
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/ResetPartitionTimeIntegrationTest.java:92]: Why are we running this test parameterized?

File: streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java [hunk 7]
```diff
@@ -145,7 +144,7 @@ private interface MetricsVerifier {
     }
 
     @ParameterizedTest
-    @ValueSource(strings = {StreamsConfig.AT_LEAST_ONCE, StreamsConfig.EXACTLY_ONCE, StreamsConfig.EXACTLY_ONCE_V2})
+    @ValueSource(strings = {StreamsConfig.AT_LEAST_ONCE, StreamsConfig.EXACTLY_ONCE_V2})
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java:147]: Why are we running this test parameterized

File: streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java [hunk 8]
```diff
@@ -171,15 +170,16 @@ private Properties streamsConfig(final String processingGuarantee, final TestInf
         final String safeTestName = safeUniqueTestName(testInfo);
         streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "test-application-" + safeTestName);
         streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
-        streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.Integer().getClass());
-        streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.IntegerSerde.class);
+        streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java:174]: Side improvement

File: streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java [hunk 9]
```diff
@@ -171,15 +170,16 @@ private Properties streamsConfig(final String processingGuarantee, final TestInf
...
         streamsConfiguration.put(StreamsConfig.METRICS_RECORDING_LEVEL_CONFIG, Sensor.RecordingLevel.DEBUG.name);
         streamsConfiguration.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, processingGuarantee);
         streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getPath());
         streamsConfiguration.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
         return streamsConfiguration;
     }
 
+    @SuppressWarnings("deprecation")
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java:182]: We are using deprecated `TimeWindows.of()` below so need this -- should be cleanup up soon with the PR which removes this deprecated method.

File: streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java (hunk 2 above)
Comment (chia7712) [reply streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java:179]: It seems the at-least-once was added by #11283

Maybe we can have a follow-up to rename this class or move the at-least-once tests out of this class.

Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java:179]: Seems this is a mess -- the original PR actually also added "early returns" for some test... Guess we need to do more follow up cleanup.

Seem one root cause if, that we have a centralized parameter originally, but now a per-test one.

But should be its own PR to do this cleanup, after this one was merged.

File: streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java (hunk 7 above)
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java:147]: I agree we probably don't need a parameterized test here and on the `ResetPartitionTimeIntegrationTest` above you called out, but maybe we can address in a followup PR

**Review by bbejeck — APPROVED** [review]
Thanks for cleanup @mjsax - LGTM

**Review by chia7712 — APPROVED** [review]

**Comment by mjsax** [comment]
Thanks! Filed https://issues.apache.org/jira/browse/KAFKA-17558

**Pull request merged**